pkg/controller/template/render.go

```diff
@@ -687,3 +711,72 @@ func isOpenShiftManagedDefaultLB(cfg RenderConfig) bool {
 	}
 	return false
 }
+
+func isCloudCredentialCapDisabled() (bool, error) {
+	restCfg, err := rest.InClusterConfig()
+	if err != nil {
+		klog.Errorln("failed to get rest config from in cluster data,", err, "trying with install config")
+		return checkInstallConfigForCCMODisabled()
+	}
+
+	cl, err := openshiftconfigclientset.NewForConfig(restCfg)
+	if err != nil {
+		return false, fmt.Errorf("failed to create clientset from rest config, %w", err)
+	}
+
+	cv, err := cl.ConfigV1().ClusterVersions().Get(context.Background(), "version", metav1.GetOptions{})
+	if err != nil {
+		return false, fmt.Errorf("failed to get clusterversion, %w", err)
+	}
+
+	knownCaps := sets.New[configv1.ClusterVersionCapability](cv.Status.Capabilities.KnownCapabilities...)
+	if !knownCaps.Has(configv1.ClusterVersionCapabilityCloudControllerManager) {
+		return false, nil
+	}
+
+	enabledCaps := sets.New[configv1.ClusterVersionCapability](cv.Status.Capabilities.EnabledCapabilities...)
+	if enabledCaps.Has(configv1.ClusterVersionCapabilityCloudControllerManager) {
+		return false, nil
+	}
+
+	return true, nil
+}
+
+func checkInstallConfigForCCMODisabled() (bool, error) {
```

Comment:
can you add some comments on each of these funcs as to what they do? for maintianability purposes